### DISCUSSION TITLE
How to customize content assist in XML editor

### DISCUSSION THREAD

**Discussion by arysin** [mickaelistria/eclipse-bluesky#983] [Q&A]
Hi I need to add a content assist in one of the attributes in XML elements.
It's a POS tag for a word so I'd need to generate it dynamically. I can write Java code and I am a bit familiar with Eclipse plugin development but would appreciate if somebody can point me on which part of [wildwebdeveloper](https://github.com/eclipse/wildwebdeveloper) I'd need to tweak.

**Comment by mickaelistria** [comment]
You can either create a LemMinX extension (independently of Eclispe IDE) and then add this extension (jar files) to the LemMinX instead started by Wild Web Developer using dedicated extension point from Wild Web Developer.
Or you can contribute a content assist processor directly to the Eclipse Generic Editor (used by Wild Web Developer) independently of LemMinX. This processor will be added to the existing ones.

**Reply by angelozerr** [reply]
I suggest really to write à lemminx extension by implementing an ICompletionParticipant because you will access to the DOMDocument and in your case DOMAttr.

Once you will create a jar you will need to register it with the wwd lemminx extension point. See m2e to have a sample which register the lemminx maven extension point.

The other benefit by implementing à lemminx extension is that you will able to use your feature in another lsp client like vscode, sublime etc

**Comment by arysin** [comment]
Thanks a lot for your help guys!
I did manage to create (sample) lemminx extension and my unit tests trigger it, and I was able to create Eclipse plugin for the extension point and I see it in plugins in Eclipse About dialog.
But I don't see it being created/triggered in Eclipse (nothing in the log).
Would it be too much to ask to take a look?
Source: https://github.com/arysin/lemminx_bruk
The generated plugin: https://github.com/arysin/lemminx_bruk/releases/download/v0.0.1/lemminx_bruk_1.0.0.SNAPSHOT.jar

**Comment by angelozerr** [comment]
See https://github.com/eclipse-m2e/m2e-core/blob/4e7ba8113e8cff1d84a05918addd8c27a108d591/org.eclipse.m2e.editor.lemminx/plugin.xml#L7

You need to register your leminx JAR with wwd lemminx extension point. It should look like this:

```xml
<extension
         point="org.eclipse.wildwebdeveloper.xml.lemminxExtension">
      <jar
            path="lemminx_bruk.jar>
      </jar>
</extension>
```

**Reply by arysin** [reply]
Thank you, that helped!
